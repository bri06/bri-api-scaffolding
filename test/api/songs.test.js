const request = require('supertest');
const app = require('../../api');

describe('Components/Songs', () => {
  it('responds with json', () => request(app)
    .get('/api/v1/songs')
    .expect(200)
    .then(response => expect(response.body).toMatchSnapshot()));

  it('responds with song detail', () => {
    const expectedSong = {
      id: 1, title: 'Song 1', artist: 'Artist 1', year: '2020',
    };
    request(app)
      .get('/api/v1/songs/1')
      .expect(200)
      .then(response => expect(response.body).toEqual(expectedSong));
  });
});
