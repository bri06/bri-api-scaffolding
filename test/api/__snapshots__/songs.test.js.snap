// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Songs responds with json 1`] = `
Array [
  Object {
    "artist": "Artist 1",
    "id": 1,
    "title": "Song 1",
    "year": "2020",
  },
  Object {
    "artist": "Artist 2",
    "id": 2,
    "title": "Song 2",
    "year": "2020",
  },
  Object {
    "artist": "Artist 3",
    "id": 3,
    "title": "Song 3",
    "year": "2020",
  },
  Object {
    "artist": "Artist 4",
    "id": 4,
    "title": "Song 4",
    "year": "2020",
  },
  Object {
    "artist": "Artist 5",
    "id": 5,
    "title": "Song 5",
    "year": "2020",
  },
  Object {
    "artist": "Artist 6",
    "id": 6,
    "title": "Song 6",
    "year": "2020",
  },
  Object {
    "artist": "Artist 7",
    "id": 7,
    "title": "Song 7",
    "year": "2020",
  },
  Object {
    "artist": "Artist 8",
    "id": 8,
    "title": "Song 8",
    "year": "2020",
  },
  Object {
    "artist": "Artist 9",
    "id": 9,
    "title": "Song 9",
    "year": "2020",
  },
]
`;
